packages/gatsby/src/redux/reducers/tasks.js

```diff
@@ -0,0 +1,61 @@
+// TODO: Ensure this is not persisted to disk since tasks are ephemeral
+const initialState = {
+  tasks: new Map(),
+}
+
+const runTask = task => {
```

Comment:
Still unclear if queue needs to be in core or workers, so no queue at the moment

Reply:
I initially was thinking of queue in core, but came around - core won't have enough context to know what are optimal params for queue for given task type (concurrency etc), unless this will be part of jobs/tasks api (which would complicate things without much gain)

Reply:
the problem here that concurrency depends on the consumer, for gatsby-cloud this can be infinity and for local this can be 4

Reply:
concurrency also depends on external systems — we'll need to add something so workers can manage backpressure from external systems to not overwhelm them e.g. downloading from a small wp instance or uploading to an API with rate limits.

Reply:
that's something that the worker needs to do or the plugin. job creation is fully async and no way to communicate back to the process during a job. As it will make it a lot harder to run a worker in a sandbox environment